esiclient/tests/functional/v1/test_image.py

```diff
@@ -0,0 +1,168 @@
+
+    def test_admin_image_public_and_private(self):
+        """Tests that an admin can create a public image and set it to private
+
+        Test steps:
+        1) Admin creates public image.
+        2) User 1 can view image.
+        3) Admin sets image to private.
+        4) User 1 can no longer view image.
+        5) Admin deletes image.
+        6) (cleanup) Delete the image created in step 1.
+
+        """
+
+        image = utils.image_create(self.clients['admin'], self.image_path)
+        self.addCleanup(utils.image_delete,
+                        self.clients['admin'],
+                        image['name'],
+                        fail_ok=True)
+
+        image_show = utils.image_show(self.clients['project1-member'],
+                                      image['name'])
+        self.assertEqual(image['id'], image_show['id'])
+
+        utils.image_set(self.clients['admin'],
+                        image['name'], '--private')
+        self.assertRaises(exceptions.CommandFailed,
+                          utils.image_show,
+                          self.clients['project1-member'],
+                          image['name'])
+
+        utils.image_delete(self.clients['admin'], image['name'])
+
+    def test_nonadmin_image_public_exception(self):
+        """Tests that a non-admin cannot create a public image
+
+        Test steps:
+        1) User 1 fails to create a public image.
+
+        """
+
+        self.assertRaises(exceptions.CommandFailed,
+                          utils.image_create,
+                          self.clients['project1-member'],
+                          self.image_path)
+
+    def test_nonadmin_image_private(self):
+        """Tests that a non-admin can create a private image
+
+        Test steps:
+        1) User 1 creates private image.
+        2) User 2 fails to view image.
+        3) User 1 deletes image.
+        4) (cleanup) Delete the image created in step 1.
+
+        """
+
+        image = utils.image_create(self.clients['project1-member'],
+                                   self.image_path,
+                                   visibility='private')
+        self.addCleanup(utils.image_delete,
+                        self.clients['admin'],
+                        image['name'],
+                        fail_ok=True)
+
+        self.assertRaises(exceptions.CommandFailed,
+                          utils.image_show,
+                          self.clients['project2-member'],
+                          image['id'])
+
+        utils.image_delete(self.clients['project1-member'], image['name'])
+
+    def test_nonadmin_image_shared(self):
+        """Tests that a non-admin can share a private image
+
+        Test steps:
+        1) User 1 creates private image.
+        2) User 2 fails to view image.
+        3) User 1 sets image visibility to shared.
+        4) User 1 shares image with user 2.
+        5) User 2 can view image.
```

Comment:
I remember user2 should run "image set --accept <image ID>" before it can show that image. IDK why we don't need this step here.